### PR TITLE
Dev/adrv9002 unit test

### PR DESCRIPTION
**Pull request by nunojsa**
# Description

This PR adds new tests/functionality to adrv9002 iio driver. It also proposes some fixes on the common conftest which are explained in the commit messages. 

- [ * ] Bug fix (non-breaking change which fixes an issue)
- [ * ] New feature (non-breaking change which adds functionality)

The tests were done on a real adrv9002 platform and confirmed to pass...

* Hardware: adrv9002 evaluation board
* OS: linux

### CODE REVIEW

**Comment by tfcollins** [comment]
@nunojsa Did you want me to add a profiles directory?

-Travis

**Comment by nunojsa** [comment]
> @nunojsa Did you want me to add a profiles directory?
> 
> -Travis

:facepalm: I did create the directory with a couple of profiles. Just forgot to commit them. Will wait a bit for some other things that I might need to change and include it in v2 of the PR

**Comment by tfcollins** [comment]
Can you also run the pre-commit checks (invoke precommit). I think mypy will complain about the profile property definition.

File: test/conftest.py [hunk 1]
```diff
@@ -262,6 +266,43 @@ def attribute_single_value_pow2(classname, devicename, attr, max_pow, tol, repea
...
+            assert bi.dev_interface(depends[p], p, tol) <= tol
+    for _ in range(repeats):
+        for val in values:
+            if isinstance(val, str):
+                assert bi.dev_interface(val, attr, 0)
+            else:
+                assert bi.dev_interface(val, attr, tol) <= tol
+
+
+def attribute_profile_write(classname, devicename, attr, file):
```
Comment (tfcollins) [inline test/conftest.py:295]: Can we make this a little more generic? Maybe call it attribute_write_only_str or something.

File: adi/adrv9002.py [hunk 2]
```diff
@@ -113,17 +113,14 @@ def __init__(self, uri=""):
...
         rx_tx.__init__(self)
 
-    @property
-    def profile(self):
-        """Load profile file. Provide path to profile file to attribute"""
-        return self._get_iio_dev_attr("profile_config")
-
-    @profile.setter
     def profile(self, value):
         with open(value, "r") as file:
```
Comment (tfcollins) [inline adi/adrv9002.py:117]: Need a docstring

**Comment by nunojsa** [comment]
Changes in v2:

- Run `pre-commit` and fix the reported failure;

- Added docstring to  `write_profile`;

- Renamed `attribute_profile_write` to `attribute_write_only_str`;

- Fixed adrv9002_examples.py since thereS is no rx_sample_rate attribute on the adrv9002 class;

- Added adrv9002 profiles folder.

**Comment by tfcollins** [comment]
Look great! Thanks

-Travis

**Pull request merged**